sw/device/tests/spi_host_irq_test.c

```diff
@@ -73,7 +73,26 @@ static status_t external_isr(void) {
                                    (dif_rv_plic_irq_id_t)
                                        kTopEarlgreyPlicIrqIdSpiHost0Error);
 
-  TRY(dif_spi_host_irq_acknowledge(&spi_host, irq_fired));
+  // Clear or Disable the interrupt as appropriate.
+  dif_irq_type_t irq_type = kDifIrqTypeEvent;
+  TRY(dif_spi_host_irq_get_type(&spi_host, irq_fired, &irq_type));
+  switch (irq_type) {
+    case kDifIrqTypeEvent:
+      TRY(dif_spi_host_irq_acknowledge(&spi_host, irq_fired));
+      break;
+    case kDifIrqTypeStatus:
+      // As the event interrupt aggregates the different events, each event has
+      // their own independent disable/mask bits (CSR.EVENT_ENABLE.x). However,
+      // we need to mask the aggregated interrupt here, and each test can handle
+      // unmasking it when it has cleared the cause or masked the individual
+      // component.
+      TRY(dif_spi_host_irq_set_enabled(&spi_host, irq_fired,
```

Comment:
Doesn't the IRQ need to be acknowledged here as well?

Reply:
No, status types are read-only; there is no acknowledge verb for them (i.e. nothing to clear at the IP-level).

They need to either have their condition handled directly in the ISR, or if that work needs to be deferred, they need to be masked before giving the plic the ack / complete. This code does the latter.